hw/mcu/nordic/nrf5340_net/src/nrf5340_net_clock.c

```diff
@@ -75,3 +75,64 @@ nrf5340_net_clock_hfxo_release(void)
 
     return stopped;
 }
+
+/**
+ * Request low frequency clock source change.
+ *
+ * @return int 0: clock source was already as requested. 1: clock source
+ *         changed.
+ */
+int
+nrf5340_net_set_lf_clock_source(uint32_t clksrc)
+{
+    uint32_t regmsk;
+    uint32_t regval;
+
+    regmsk = CLOCK_LFCLKSTAT_STATE_Msk | CLOCK_LFCLKSTAT_SRC_Msk;
+    regval = CLOCK_LFCLKSTAT_STATE_Running << CLOCK_LFCLKSTAT_STATE_Pos;
+
+    regval |= clksrc << CLOCK_LFCLKSTAT_SRC_Pos;
+
+    /* Check if this clock source isn't already running */
+    if ((NRF_CLOCK_NS->LFCLKSTAT & regmsk) == regval) {
+        return 0;
+    }
+
+    /*
+     * Request HFXO if LFSYNTH is going to be set as source. If LFSYNTH is going to be
+     * replaced with other source, release HFXO.
+     */
+    if (clksrc == CLOCK_LFCLKSTAT_SRC_LFSYNT) {
+        if ((NRF_CLOCK_NS->HFCLKSTAT & CLOCK_HFCLKSTAT_STATE_Msk) !=
+            (CLOCK_HFCLKSTAT_STATE_Running << CLOCK_HFCLKSTAT_STATE_Pos)) {
+            NRF_CLOCK_NS->EVENTS_HFCLKSTARTED = 0;
+            nrf5340_net_clock_hfxo_request();
+            while (1) {
+                if ((NRF_CLOCK_NS->EVENTS_HFCLKSTARTED) != 0) {
+                    break;
+                }
+            }
+        } else {
+            nrf5340_net_clock_hfxo_request();
+        }
+    } else if ((NRF_CLOCK_NS->LFCLKSTAT & CLOCK_LFCLKSTAT_SRC_Msk) ==
+               CLOCK_LFCLKSTAT_SRC_LFSYNT) {
+        nrf5340_net_clock_hfxo_release();
+    }
+
+    NRF_CLOCK_NS->TASKS_LFCLKSTOP = 1;
+    NRF_CLOCK_NS->EVENTS_LFCLKSTARTED = 0;
+    NRF_CLOCK_NS->LFCLKSRC = clksrc;
+    NRF_CLOCK_NS->TASKS_LFCLKSTART = 1;
```

Comment:
lfclk source can be changes without restarting lfclk:
`The LFCLK source can be configured at any time (for instance, when the LFCLK has already been started). The content of the LFCLKSRC register only takes effect when the LFCLKSTART task is triggered.`

so we can only start lfclk once on system startup, then this API can change source at any time only taking care of hfxo when switching to/from lfsynth